src/schema-builder/RdbmsSchemaBuilder.ts

```diff
@@ -98,17 +95,24 @@ export class RdbmsSchemaBuilder implements SchemaBuilder {
         }
     }
 
+    /**
+     * If the schema contains views, create the typeorm_metadata table if it doesn't exist yet
+     */
+    async createMetadataTableIfNecessary(): Promise<void> {
+        if (!this.queryRunner) this.queryRunner = this.connection.createQueryRunner("master");
```

Comment:
why did you add this check? it looks like this.queryRunner is created in all places where we use this method. Having it just for type is super confusing. But if its really needed (but I don't think so) it should also be properly released.